### ISSUE TITLE
Crashing on "Hello Probo"

### ISSUE THREAD

**Issue by prscoelho** [big-ab-games/robo-instructus#240]
Hi. I'm getting a crash on the first level of act 4, the bot walks off the edge and then the game crashes.

error.log
```
2020-08-22T16:27:35Z (compute) nothing to receive -    0: robo_instructus_compute::error::panic_hook::{{closure}}
             at /root/project/robo-instructus/compute/src/error.rs:38
   1: std::panicking::rust_panic_with_hook
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/panicking.rs:490
   2: rust_begin_unwind
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/panicking.rs:388
   3: core::panicking::panic_fmt
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libcore/panicking.rs:101
   4: core::option::expect_failed
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libcore/option.rs:1264
   5: core::option::Option<T>::expect
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libcore/option.rs:349
      robo_instructus_compute::world::World::compute
             at /root/project/robo-instructus/compute/src/world/mod.rs:519
   6: robo_instructus_compute::compute::compute_world_state
             at /root/project/robo-instructus/compute/src/compute/mod.rs:1097
   7: robo_instructus_compute::compute::compute_code_world_state
             at /root/project/robo-instructus/compute/src/compute/mod.rs:1028
      robo_instructus_compute::compute::compute_state
             at /root/project/robo-instructus/compute/src/compute/mod.rs:648
   8: robo_instructus_compute::compute::start
             at /root/project/robo-instructus/compute/src/compute/mod.rs:383
   9: robo_instructus_compute::compute::start_in_new_thread::{{closure}}
             at /root/project/robo-instructus/compute/src/compute/mod.rs:93
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/sys_common/backtrace.rs:130
  10: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/thread/mod.rs:475
      <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/panic.rs:318
      std::panicking::try::do_call
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/panicking.rs:297
      std::panicking::try
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/panicking.rs:274
      std::panic::catch_unwind
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/panic.rs:394
      std::thread::Builder::spawn_unchecked::{{closure}}
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/thread/mod.rs:474
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libcore/ops/function.rs:232
  11: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/liballoc/boxed.rs:1076
      <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/liballoc/boxed.rs:1076
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/sys/unix/thread.rs:87
  12: start_thread
  13: __GI___clone
```

Code that crashed:
```
fun ask_scan()
    transmit(0, 0) # ask bot if I can move forward
    loop
        var rx = receive(1)
        if rx is not -20000
            pause()
            return rx
fun move_forward() # move forward and transmit order
    transmit(0, 1)
    robo_forward()
fun move_left()
    transmit(0, 2)
    robo_left()
loop # main
    var scan = ask_scan()
    if scan > 0
        move_forward()
    else
        move_left()
#########
fun do_order(order)
    if order is -20000
        return
    if order is 0
        var scan = probo_scan()
        transmit(1, scan)
    else if order is 1
        probo_forward()
    else if order is 2
        probo_left()
loop # main
    var order = receive(0)
    do_order(order)
```
Was just the first thing I thought to try, I see now that the bot scans current tile and not the one in front of it so the strat doesn't work... Thanks for the game by the way, it's mind blowingly good.

**Comment by alexheretic** [comment]
Thanks for the report and well done for getting so far in the game! 

This bug is caused by the robot falling clearing some internal transmission state that the still running probe expects to exist. Fix incoming...

**Comment by alexheretic** [comment]
Fixed in 1.29.1

**Comment by prscoelho** [comment]
Thank you so much for the quick fix!
